tests/main/cgroup-tracking/task.yaml

```diff
@@ -0,0 +1,104 @@
+summary: Each snap app and hook is tracked via cgroups
+# Ubuntu 14.04's special version of systemd doesn't have StartTransientUnit API.
+systems: [-ubuntu-14.04-*]
+details: |
+    This test creates a snap process that suspends itself and ensures that it
+    placed into the appropriate hierarchy.
+prepare: |
+    # This feature depends on the release-app-awareness feature
+    snap set core experimental.refresh-app-awareness=true
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-sh
+
+restore: |
+    snap unset core experimental.refresh-app-awareness
+
+execute: |
+    # TODO: add a variant that runs as an user
+    # TODO: add a variant that uses a service
+
+    # This test varies between Ubuntu 16.04, Ubuntu 18.04 and Fedora 31.
+    # This combination exercises each of the three cases below.
```

Comment:
I know it is documented in the code, but documenting how they differ would make verifying correctness here easier.

Reply:
This is done now.